### DISCUSSION TITLE
I cannot style the html data from the API

### DISCUSSION THREAD

**Discussion by miracim** [tailwindlabs/tailwindcss#14419] [Help]
I'm using NextJS 14 and I'm pulling html from an API, I'm printing it with the parser and there are tailwind classes in it. navigation and footer are static, that is, the files I added myself, while tailwindcss is running in these files, when I print the html from the API, styling is not applied. actually it is applied, but I can say that some of them are applied, for example, all media queries do not work. md:flex, xl:grid and no responsive classes like that work. What can i do? i tried to use safelist but it didn't work! 

example issue: https://stackoverflow.com/questions/73186053/next-js-the-style-of-the-tailwind-class-i-get-from-api-doesnt-apply-on-my-elem

**Comment by RobinMalfait** [comment]
Hey!

Tailwind works at build-time, so all classes coming from an API request at runtime won't be seen by the Tailwind compiler because at that point there is only a CSS file available.

In those cases, using the safelist is a solution to the problem (https://tailwindcss.com/docs/content-configuration#safelisting-classes).

But since you mentioned that it doesn't work, can you share a minimal reproduction that we can take a look at?
